v5/ooapi.v5.core.UnitTests/Repositories/BaseRepositoryTests.cs

```diff
@@ -0,0 +1,121 @@
+
+        var dataRequestParameters = new DataRequestParameters { PrimaryCodeSearch = "123" };
+        var set = new List<Foo>()
+        {
+            new Foo() { PrimaryCode = "123" },
+            new Foo() { PrimaryCode = "456" }
+        }.AsQueryable();
+
+        // Act
+        var result = repository.GetAllOrderedBy(dataRequestParameters, set);
+
+        // Assert
+        Assert.IsInstanceOf<Pagination<Foo>>(result);
+        Assert.That(result.Items.Count(), Is.EqualTo(1));
+    }
+
+    [Test]
+    public void GetAllOrderedBy_WithFilters_ReturnsFilteredSet()
+    {
+        // Arrange
+        var set = new List<Foo>()
+        {
+            new Foo() { PrimaryCode = "123", PrimaryCodeType = "x-test"},
+            new Foo() { PrimaryCode = "456", PrimaryCodeType = "not-x-test"}
+        }.AsQueryable();
+
+        var dataRequestParameters = new DataRequestParameters { Filters = new Dictionary<string, object> { { "PrimaryCodeType", "x-test" } } };
+        var repository = GetRepository<Foo>();
+
+        // Act
+        var result = repository.GetAllOrderedBy(dataRequestParameters, set);
+
+        // Assert
+        Assert.IsInstanceOf<Pagination<Foo>>(result);
+        Assert.That(result.Items.Count, Is.EqualTo(1));
+    }
+
+    [Test]
+    public void GetAllOrderedBy_WithNoParameters_ReturnsAll()
+    {
+        // Arrange
+        var dataRequestParameters = new DataRequestParameters();
+        var set = new List<Foo>()
+        {
+            new Foo() { PrimaryCode = "123", PrimaryCodeType = "x-test"},
+            new Foo() { PrimaryCode = "456", PrimaryCodeType = "not-x-test"}
+        }.AsQueryable();
+        var repository = GetRepository<Foo>();
+
+        // Act
+        var result = repository.GetAllOrderedBy(dataRequestParameters, set);
+
+        // Assert
+        Assert.IsInstanceOf<Pagination<Foo>>(result);
+        Assert.That(result.Items.Count, Is.EqualTo(2));
+    }
+
+    [Test]
+    public void GetAllOrderedBy_WithNoSetSupplied_ReturnsSetFromDbContext()
+    {
+        // Arrange
+        var dataRequestParameters = new DataRequestParameters();
+        var dbContextSet = new List<Foo>()
+        {
+            new Foo() { PrimaryCode = "123", PrimaryCodeType = "x-test"},
+            new Foo() { PrimaryCode = "456", PrimaryCodeType = "not-x-test"}
+        }.AsQueryable();
+        var repository = GetRepository<Foo>(dbContextSet);
+
+        // Act
+        var result = repository.GetAllOrderedBy(dataRequestParameters);
+
+        // Assert
+        Assert.IsInstanceOf<Pagination<Foo>>(result);
+    }
+
+    private static BaseRepository<T> GetRepository<T>(IQueryable<T>? setForDbContext = null) where T : class
+    {
+        var dbContext = Substitute.For<ICoreDbContext>();
+        if (setForDbContext is not null)
+        {
+            dbContext.Set<T>().AsQueryable().Returns(setForDbContext);
+        }
+
+        return new BaseRepository<T>(dbContext);
+    }
+}
+
+public class Foo
```

Comment:
private class, en binnen `BaseRepositoryTests` definieren?

Reply:
Ik heb hem public gehouden want anders kan er geen Substitute van gemaakt worden in GetRepository. Heb hem wel binnen de `BaseRepositoryTests` geplaatst.